### PR TITLE
RethinkDB Test

### PR DESCRIPTION
**Pull request by mlucy**
Requires the version of `clj-rethinkdb` at https://github.com/mlucy/clj-rethinkdb/tree/mlucy_error_types .

### CODE REVIEW

**Comment by mlucy** [comment]
I've never written clojure before, so apologies in advance if it's unidiomatic.

File: rdb/project.clj [hunk 1]
```diff
@@ -0,0 +1,17 @@
+(defproject mongodb "0.1.0-SNAPSHOT"
+  :description "RethinkDB Jepsen Tests"
+;  :url "http://github.com/rethinkdb/jepsen"
```
Comment (mlucy) [inline rdb/project.clj]: What should go here do you think?  All the other toplevel directories had `example.com/FIXME` here.

File: rdb/src/rdb/core.clj [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+  "Rethinkdb (ignores version)."
+  (reify db/DB
+
+    (setup! [_ test node]
+      (debian/install [:libprotobuf7 :libicu48 :psmisc])
+      (info node "Starting...")
+      ;; TODO: detect server failing to start.
+      (c/ssh* {:cmd "killall rethinkdb bash"})
+      (c/ssh* {:cmd "
+# Add a 100ms delay to help find edge cases.
```
Comment (mlucy) [inline rdb/src/rdb/core.clj]: What's your policy on code that's sometimes useful when fiddling but isn't being actively used?  Should I just cut these commented-out lines?

File: rdb/src/rdb/document_cas.clj [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+(defn r   [_ _] {:type :invoke, :f :read})
+(defn cas [_ _] {:type :invoke, :f :cas, :value [(rand-int 5) (rand-int 5)]})
+
+(defn safe-test
+  "Document-level compare and set with safe settings."
+  []
+  ;; This is the only safe read/write mode.  Changing either of
+  ;; these (or turning on soft durability) may produce a
+  ;; non-linearizable history.
+  (test- "document {:write_acks 'majority' :read_mode 'majority'}"
```
Comment (mlucy) [inline rdb/src/rdb/document_cas.clj]: I was in fact able to get a non-linearizable history changing these, so the test is actually testing something.

File: rdb/project.clj (hunk 1 above)
Comment (aphyr) [inline rdb/project.clj]: You can use the jepsen repo URL if you like, but it's not mandatory. I'm more concerned that the project name is "mongodb". I suggest "jepsen.rethinkdb".

File: rdb/src/rdb/core.clj [hunk 4]
```diff
@@ -0,0 +1,135 @@
+(ns rdb.core
```
Comment (aphyr) [inline rdb/src/rdb/core.clj]: For inclusion in the main Jepsen repo, let's call this jepsen.rethinkdb

File: rdb/src/rdb/core.clj (hunk 2 above)
Comment (aphyr) [inline rdb/src/rdb/core.clj]: `#` isn't a comment in Clojure--did you mean `;`? I'm fine with commented out stuff though. :)

Comment (aphyr) [inline rdb/src/rdb/core.clj]: Oh wait this is all one massive bash script? I'm sorry, I didn't realize. Um... how about moving this out to resources/setup.sh, and slurping it in via (slurp (io/resource "setup.sh"))?

File: rdb/src/rdb/core.clj [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+base=http://download.rethinkdb.com/apt/pool
+version=/precise/main/r/rethinkdb/rethinkdb_2.1.2~0precise_amd64.deb
+if [[ ! -f /root/rdb.deb ]]; then
+  curl $base/$version > /root/rdb.deb
+  dpkg -i /root/rdb.deb
+fi
+nohup /usr/bin/rethinkdb \\
+  --bind all \\
+  -n `hostname` \\
+  -j n1:29015 -j n2:29015 -j n3:29015 -j n4:29015 -j n5:29015 \\
```
Comment (aphyr) [inline rdb/src/rdb/core.clj]: These should be parameterized by (:nodes test)

File: rdb/src/rdb/document_cas.clj (hunk 3 above)
Comment (aphyr) [inline rdb/src/rdb/document_cas.clj]: Cool!

**Comment by aphyr** [comment]
Things look pretty good to me! Just a few minor namespace changes and some small refactors and we'll be good to merge. Thank you!

**Comment by mlucy** [comment]
Sorry for the delay!

I think I addressed everything you mentioned except for the namespace issue.  I don't really understand how clojure package management works; I fiddled with it for a bit but couldn't figure out how to change the namespaces to `rethinkdb.X` without it conflicting with the `rethinkdb` driver package.

**Comment by mlucy** [comment]
(To clarify a bit, I couldn't figure out how to e.g. make the first like of `core.clj` be `(ns rethinkdb.core` without the `[rethinkdb.core :refer [connect close]]` on line 22 breaking.)

**Comment by aphyr** [comment]
Don't worry about it! Thanks so much for getting all this put together. :)

**Pull request merged**